File: backend/internal/dto/classroom_join_dto.go
package dto

import "time"

// ===== REQUEST DTOs =====

type JoinClassroomRequest struct {
	InvitationCode string `json:"invitation_code" binding:"required"`
}

type UploadWhitelistRequest struct {
	StudentCodes []string `json:"student_codes" binding:"required,min=1"`
}

// ===== RESPONSE DTOs =====

type ClassroomPreviewResponse struct {
	ID           string `json:"id"`
	Name         string `json:"name"`
	Lecturer     string `json:"lecturer"`
	StudentCount int    `json:"student_count"`
	MaxStudents  int    `json:"max_students"`
}

type JoinClassroomResponse struct {
	ClassroomID string `json:"classroom_id"`
	Status      string `json:"status"` // "approved" or "pending"
	Message     string `json:"message"`
}

type JoinRequestResponse struct {
	ID          string    `json:"id"`
	StudentCode string    `json:"student_code"`
	FullName    string    `json:"full_name"`
	Email       string    `json:"email"`
	Status      string    `json:"status"`
	CreatedAt   time.Time `json:"created_at"`
}

File: frontend/src/dtos/classroom-join-dto.ts
// ==================== REQUEST DTOs ====================

export interface JoinClassroomRequest {
  invitationCode: string;
}

export interface UploadWhitelistRequest {
  studentCodes: string[];
}

// ==================== RESPONSE DTOs ====================

export interface ClassroomPreviewResponse {
  id: string;
  name: string;
  lecturer: string;
  studentCount: number;
  maxStudents: number;
}

export interface JoinClassroomResponse {
  classroomId: string;
  status: string; // "approved" or "pending"
  message: string;
}

export interface JoinRequestResponse {
  id: string;
  studentCode: string;
  fullName: string;
  email: string;
  status: string;
  createdAt: string;
}
